src/thermo/CoverageDependentSurfPhase.cpp

```diff
@@ -0,0 +1,772 @@
+            grt[k] += tmp;
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getPureGibbs(doublereal* g) const
+{
+    getGibbs_RT(g);
+    for (size_t k = 0; k < m_kk; k++) {
+        g[k] *= RT();
+    }
+}
+
+void CoverageDependentSurfPhase::getStandardChemPotentials(doublereal* mu0) const
+{
+    _updateThermo();
+    copy(m_chempot.begin(), m_chempot.end(), mu0);
+    if (m_has_ref_coverage) {
+        double tmp = RT() * -log(m_theta_ref);
+        for (size_t k = 0; k < m_kk; k++) {
+            mu0[k] += tmp;
+        }
+    }
+}
+
+// Functions calling partial molar thermodyanmic properties----------------
+
+void CoverageDependentSurfPhase::getPartialMolarEnthalpies(doublereal* hbar) const
+{
+    _updateThermo();
+    copy(m_enthalpy.begin(), m_enthalpy.end(), hbar);
+}
+
+void CoverageDependentSurfPhase::getPartialMolarEntropies(doublereal* sbar) const
+{
+    _updateThermo();
+    copy(m_entropy.begin(), m_entropy.end(), sbar);
+    if (m_has_ref_coverage) {
+        for (size_t k = 0; k < m_kk; k++) {
+            sbar[k] -= GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber) / m_theta_ref);
+        }
+    } else {
+        for (size_t k = 0; k < m_kk; k++) {
+            sbar[k] -= GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber));
+        }
+    }
+}
+
+void CoverageDependentSurfPhase::getPartialMolarCp(doublereal* cpbar) const
+{
+    _updateThermo();
+    copy(m_heatcapacity.begin(), m_heatcapacity.end(), cpbar);
+}
+
+
+void CoverageDependentSurfPhase::getChemPotentials(doublereal* mu) const
+{
+    _updateThermo();
+    copy(m_chempot.begin(), m_chempot.end(), mu);
+    if (m_has_ref_coverage) {
+        for (size_t k = 0; k < m_kk; k++) {
+            mu[k] += RT() * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber) / m_theta_ref);
+        }
+    } else {
+        for (size_t k = 0; k < m_kk; k++) {
+            mu[k] += RT() * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber));
+        }
+    }
+}
+
+// Functions calculating mixture thermodyanmic properties--------------------------
+
+doublereal CoverageDependentSurfPhase::enthalpy_mole() const
+{
+    if (m_n0 <= 0.0) {
+        return 0.0;
+    }
+    _updateThermo();
+    return mean_X(m_enthalpy);
+}
+
+doublereal CoverageDependentSurfPhase::entropy_mole() const
+{
+    _updateThermo();
+    doublereal entropy = 0.0;
+    if (m_has_ref_coverage) {
+        for (size_t k = 0; k < m_kk; k++) {
+            entropy += moleFraction(k) * (m_entropy[k] -
+                GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber) / m_theta_ref));
+        }
+    } else {
+        for (size_t k = 0; k < m_kk; k++) {
+            entropy += moleFraction(k) * (m_entropy[k] -
+                GasConstant * log(std::max(concentration(k) * size(k)/m_n0, 
+                SmallNumber)));
+        }
+    }
```

Comment:
The way `m_theta_ref` appears to be used, it effectively has a default value of 1, right? That could be used to eliminate the `m_has_ref_coverage` variables and a number of `if (m_has_ref_coverage)` checks like this one.